pkg/start/start_integration_test.go

```diff
@@ -563,7 +563,7 @@ func TestIntegrationCVO_gracefulStepDown(t *testing.T) {
 	options.NodeName = "test-node"
 	controllers := options.NewControllerContext(cb)
 
-	worker := cvo.NewSyncWorker(&mapPayloadRetriever{}, cvo.NewResourceBuilder(cfg, cfg, nil, nil), 5*time.Second, wait.Backoff{Steps: 3}, "", record.NewFakeRecorder(100), payload.DefaultClusterProfile)
+	worker := cvo.NewSyncWorker(&mapPayloadRetriever{}, cvo.NewResourceBuilder(cfg, cfg, nil, nil), 5*time.Second, wait.Backoff{Steps: 3}, "", false, record.NewFakeRecorder(100), payload.DefaultClusterProfile)
```

Comment:
I realize that these tests are all "we are not exercising TechPreviewNoUpgrade", but it feels a bit strange to be calling `NewControllerContext` two lines up, with its Kube-API call to check the feature gates and store the value for the CVO and stopper, and then here assume it's `false` instead of...  I guess two options:

a. Reach into `controllers` and pull `includeTechPreview` off of something.  Although I guess both the CVO and the stopper are holding the value in internal properties that we can't access from this package.  So maybe this one's not an option after all.
b. Shift the Kube-API call out of `NewControllerContext` and up into `Run`, and have the value passed into `NewControllerContext`.  Then here you can explicitly pass `false` into both `NewControllerContext` and `NewSyncWorker`.

Reply:
fixed in fda77fe; thanks :)  Pre-existing brittle-ness from the `SetSyncWorkerForTesting` flow can slide.